backend/multimodalexplorer/functions/search_faiss_index.py

```diff
@@ -87,7 +88,31 @@ def _process_search_query(self, search_query: dict) -> np.ndarray:
 
         return query_embedding.numpy().astype(np.float32)
 
-    def search_index(self, search_query: dict) -> list:
+    def _query_result(self, indices) -> list:
+
+        if len(indices) == 0:
+            return []
+
+        dir_path, ext = self.raw_data_file.values()
+        file_path = get_file_path(dir_path, ext)
+
+        raw_data = pd.read_csv(file_path, sep="\t")
```

Comment:
this means that you are reading the whole data at every query, which is going to be very slow. You should maintain this data in memory or at least in a memory mapped structure. pyarrow should help you with that: https://arrow.apache.org/docs/python/csv.html You shouldn't even have to drop to pandas, just use `to_table` and use pyarrow table for quick access: https://arrow.apache.org/docs/python/data.html#tables